build/future_function_templates/future.c.template

```diff
@@ -7,14 +7,23 @@
 
 {{ header_comment }}
 
+#define FUTURE_TIMEOUT_ABORT                         \
+   if (1) {                                          \
+      fflush (stdout);                               \
+      fprintf (stderr, "%s timed out\n", BSON_FUNC); \
+      fflush (stderr);                               \
+      abort ();                                      \
+   } else                                            \
+      ((void) 0)
```

Comment:
I'm curious why we use this `if (1) { ... } else` pattern in several places rather than the more common `do { ... } while (0)`?

Reply:
I think that pattern started recently from @vector-of-bool. [This article](https://gustedt.wordpress.com/2011/02/02/handling-control-flow-inside-macros/) describes the motivation.